### PR TITLE
Add User Data Shell Script Is Encoded query #1413

### LINKED ISSUE #1413: Add User Data Shell Script Is Encoded query for Ansible
### Platform
Ansible

### Provider
AWS

### Description
User Data Shell Script must be encoded

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
User Data Shell Script must be encoded

Closes #1413

### CODE REVIEW

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049026&projectid=713)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_user_data_shell_script_encoded/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049026&projectid=713&pathid=3)
[52](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_user_data_shell_script_encoded/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049026&projectid=713&pathid=1)
[83](https://github.com/Checkmarx/kics/blob/feature/add_query_ansible_user_data_shell_script_encoded/test/main_test.go#L83) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049026&projectid=713&pathid=2)

File: assets/queries/ansible/aws/user_data_shell_script_is_encoded/test/positive_expected_result.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+	{
+		"queryName": "User Data Shell Script Is Encoded",
+		"severity": "HIGH",
+		"line": 1
```
Comment (joaomartinscx) [inline assets/queries/ansible/aws/user_data_shell_script_is_encoded/test/positive_expected_result.json:5]: Despite this being correct with the engine, the line that should be identified should be 9, where `user_data` is present

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Comment by joaomartinscx** [comment]
The search key needs the resource name to distinguish different resources of the same type

**Review by joaomartinscx — APPROVED** [review]
Everything looks good to me, good job!

**Pull request merged**